src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,429 @@
+#include "encoder/encoderfdkaac.h"
+
+#include <QDir>
+#include <QStandardPaths>
+#include <QString>
+#include <QStringList>
+
+#include "engine/sidechain/enginesidechain.h"
+#include "recording/defs_recording.h"
+#include "util/logger.h"
+#include "util/sample.h"
+
+namespace {
+// recommended in encoder documentation, section 2.4.1
+const int kOutBufferBits = 6144;
+const mixxx::Logger kLogger("EncoderFdkAac");
+} // namespace
+
+EncoderFdkAac::EncoderFdkAac(EncoderCallback* pCallback)
+        : aacEncOpen(nullptr),
+          aacEncClose(nullptr),
+          aacEncEncode(nullptr),
+          aacEncInfo(nullptr),
+          aacEncoder_SetParam(nullptr),
+          m_aacAot(AOT_AAC_LC),
+          m_bitrate(0),
+          m_channels(0),
+          m_samplerate(0),
+          m_pCallback(pCallback),
+          m_library(nullptr),
+          m_pInputFifo(nullptr),
+          m_pFifoChunkBuffer(nullptr),
+          m_readRequired(0),
+          m_aacEnc(),
+          m_pAacDataBuffer(nullptr),
+          m_aacInfo(),
+          m_hasSbr(false) {
+    // Load shared library
+    // Code import from encodermp3.cpp
+    QStringList libnames;
+#ifdef __LINUX__
+    libnames << "fdk-aac";
+    libnames << "libfdk-aac.so.2";
+    libnames << "libfdk-aac.so.1";
+#elif __WINDOWS__
+    // Give top priority to libfdk-aac copied
+    // into Mixxx's installation folder
+    libnames << "libfdk-aac-2.dll";
+    libnames << "libfdk-aac-1.dll";
+
+    // Fallback and user-friendly method: use libfdk-aac
+    // provided with B.U.T.T installed in
+    // a standard location
+    QString buttFdkAacPath = buttWindowsFdkAac();
+    if (!buttFdkAacPath.isNull()) {
+        kLogger.debug() << "Found libfdk-aac at" << buttFdkAacPath;
+        libnames << buttFdkAacPath;
+    }
+
+    // Last resort choices: try versions with unusual names
+    libnames << "libfdk-aac.dll";
+    libnames << "libfdkaac.dll";
+#elif __APPLE__
```

Comment:
This will fail to find the dylib shipped with Mixxx.

Reply:
I doubt these absolute paths are needed. Shouldn't those already be in `LD_LIBRARY_PATH` if Homebrew or MacPorts are installed?

Reply:
Where will the shipped library installed? 
We need to add it at the end of this list, because it serves a fall back.

I guess the LD_LIBRARY_PATH is configured for the general case to prefer the shipped libraries which is wrong in our case.

Reply:
> Where will the shipped library installed?

`Contents/Library` inside the `.app` bundle. I presume QLibrary will search here automatically if you don't override it with an absolute path. Let's try.

Reply:
The library is now installed into the macOS bundle:
https://github.com/mixxxdj/mixxx/pull/3543/checks?check_run_id=1824825196#step:21:2718
But it will not be found with these absolute paths.

Reply:
Test build without absolute path for macOS: https://github.com/Be-ing/mixxx/actions/runs/534911158
Test build without absolute path for Windows: https://github.com/Be-ing/mixxx/actions/runs/534918935